5-design-patterns/homework/src/Client.ts

```diff
@@ -0,0 +1,71 @@
+import { ShipmentParams } from './interfaces';
+import { Shipment } from "./Shipment";
+import { Letter } from "./Letter";
+import { Package } from "./Package";
+import { Oversized } from "./Oversized";
+import { Shipper } from "./Shipper";
+import { AirEastShipper } from './AirEastShipper';
+import { ChicagoSprintShipper } from './ChicagoSprintShipper';
+import { PacificParcelShipper } from "./PacificParcelShipper";
+
+export class Client {
+    private readonly shipmentParams: ShipmentParams;
+    private ZIP_CODE_FIRST_NUMBERS = {
+        airEast: [1, 2, 3],
+        chicagoSprint: [4, 5, 6],
+        pacificParcel: [7, 8, 9]
+    }
+
+    constructor(shipmentParams: ShipmentParams) {
+        this.shipmentParams = shipmentParams;
+    }
+
+    getShipperDetails (): Shipper {
```

Comment:
The same here. I would encapsulate this logic of creating shipper in another place + in diagram I don't see links between client and shipper. Ideally Client should ship a shipment without deciding which shipper/shipment type should be.

Reply:
In the architecture we expect Shipment `ships` itself, so it better knows which shippper can be used